sources/packages/backend/apps/api/src/route-controllers/models/pagination.dto.ts

```diff
@@ -75,8 +87,28 @@ export class StudentAppealPendingPaginationOptionsAPIInDTO extends PaginationOpt
 
 export class ProgramsPaginationOptionsAPIInDTO extends PaginationOptionsAPIInDTO {
   @IsOptional()
-  @IsIn(["submittedDate", "programName", "credentialType"])
+  @IsIn([
+    "submittedDate",
+    "programName",
+    "locationName",
+    "programStatus",
+    "credentialType",
+  ])
   sortField?: string;
+  @IsOptional()
+  @MaxLength(PAGINATION_SEARCH_MAX_LENGTH)
+  programNameSearch?: string;
+  @IsOptional()
+  @MaxLength(PAGINATION_SEARCH_MAX_LENGTH)
+  locationNameSearch?: string;
+  @IsArray()
+  @IsOptional()
+  @Transform(({ value }) => value.split(","))
+  statusSearch?: ProgramStatus[];
```

Comment:
We should ensure that the split values are expected enum values.
Can you try the below?
```ts
@IsEnum(ProgramStatus, { each: true })
```